### ISSUE TITLE
delegate_dataset fails if zones has explicit mountpoint

### ISSUE THREAD

**Issue by ismell** [joyent/smartos-live#642]
If the zones pool has an explicit mountpoint:

    zones  mountpoint            /zones                   local

and then I try and create a zone with delegate_dataset: true I get the following error:

    root@headnode /var/tmp/zones]# vmadm create -f documents-lx.json
    Command failed: could not verify zfs dataset zones/bfff0017-6754-c2f5-de54-a5c580c9430c/data:     mountpoint cannot be inherited
    zoneadm: zone bfff0017-6754-c2f5-de54-a5c580c9430c failed to verify

Maybe vmadm needs to set the mountpoint explicitly for the delegated dataset or just not have an explicit mountpoint on the zones pool ;)

**Comment by foxx** [comment]
Having this same issue..

**Comment by ismell** [comment]
Just unset your explicit mountpoint on the zones pool.

**Comment by foxx** [comment]
Yeah that worked, but it would be good if the error was more verbose about why it failed
